signer/src/network/libp2p/event_loop.rs

```diff
@@ -155,19 +161,22 @@ pub async fn run(ctx: &impl Context, swarm: Arc<Mutex<Swarm<SignerBehavior>>>) {
                             let result = swarm
                                 .behaviour_mut()
                                 .kademlia
-                                .add_address(&peer_id, address.clone());
-
-                            match result {
-                                RoutingUpdate::Success => {
-                                    tracing::debug!(%peer_id, %address, "added peer address to kademlia");
-                                }
-                                RoutingUpdate::Failed => {
-                                    tracing::warn!(%peer_id, %address, "failed to add peer address to kademlia");
+                                .as_mut() // Returns `None` if disabled.
+                                .map(|kademlia| kademlia.add_address(&peer_id, address.clone()));
+
+                            if let Some(result) = result {
+                                match result {
+                                    RoutingUpdate::Success => {
```

Comment:
This is fine, but we can also do:
```rust
match result {
    Some(RoutingUpdate::Success) => {...}
    ...
    None => {}
}   
```

Reply:
Yeah, I was on the fence. I just found:
```
Some(RoutingUpdate::Success) => ...
Some(RoutingUpdate::Failed) => ...
Some(RoutingUpdate::Pending) => ...
None => {}
```
more annoying to the eye than:
```
RoutingUpdate::Success => ...
RoutingUpdate::Failed => ...
RoutingUpdate::Pending => ...
```
RoutingUpdate::Success